File: app/Models/Course.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Model;

class Course extends Model
{
    protected $fillable = [
        'teacher_id',
        'title',
        'description',
        'start_time',
        'end_time',
        'timezone',
        'max_participants',
        'visibility',
        'token_type_id',
        'access_code',
        'is_recurring',
        'parent_course_id',
    ];

    protected $casts = [
        'start_time' => 'datetime',
        'end_time' => 'datetime',
        'is_recurring' => 'boolean',
    ];

    public function teacher()
    {
        return $this->belongsTo(User::class, 'teacher_id');
    }

    public function tokenType()
    {
        return $this->belongsTo(TokenType::class, 'token_type_id');
    }

    public function invitations()
    {
        return $this->hasMany(CourseInvitation::class);
    }

    public function enrollments()
    {
        return $this->hasMany(Enrollment::class);
    }

    public function recordings()
    {
        return $this->hasMany(Recording::class);
    }

    public function parentCourse()
    {
        return $this->belongsTo(Course::class, 'parent_course_id');
    }

    public function childCourses()
    {
        return $this->hasMany(Course::class, 'parent_course_id');
    }
}

File: applicationKT/app/src/main/java/com/englishapp/data/model/Course.kt
package com.englishapp.data.model

import com.squareup.moshi.Json
import com.squareup.moshi.JsonClass

@JsonClass(generateAdapter = true)
data class Course(
    @Json(name = "id") val id: Long,
    @Json(name = "teacher_id") val teacherId: Long,
    @Json(name = "title") val title: String,
    @Json(name = "description") val description: String? = null,
    @Json(name = "start_time") val startTime: String,
    @Json(name = "end_time") val endTime: String,
    @Json(name = "timezone") val timezone: String = "UTC",
    @Json(name = "max_participants") val maxParticipants: Int? = null,
    @Json(name = "visibility") val visibility: String,
    @Json(name = "token_type_id") val tokenTypeId: Long? = null,
    @Json(name = "access_code") val accessCode: String? = null,
    @Json(name = "is_recurring") val isRecurring: Boolean = false,
    @Json(name = "parent_course_id") val parentCourseId: Long? = null,
    @Json(name = "token_type") val tokenType: TokenType? = null,
    @Json(name = "teacher") val teacher: User? = null,
    @Json(name = "created_at") val createdAt: String? = null,
    @Json(name = "updated_at") val updatedAt: String? = null
)
